src/views/validated-designs/server.ts

```diff
@@ -75,56 +80,176 @@ export function getHvdLandingProps(): ValidatedDesignsLandingProps | null {
+		const [product, category] = pathParts
 
-		// Guard so that we only work with valid product directories
 		if (!isProductSlug(product)) {
 			return
 		}
 
+		// We assume category config files have 3 path parts, and hvd guide configs have 4 parts
+		const isCategoryMetadata = pathParts.length === 3
+		const isHvdMetadata = pathParts.length === 4
+
+		const slug = `${product}-${category}`
 		if (isCategoryMetadata) {
 			const { title, description } = loadMetadata(
 				path.join(HVD_CONTENT_DIR, item)
 			)
-			categoryGroups[categorySlug] = {
-				slug: categorySlug,
+
+			hvdCategoryGroups.push({
+				slug,
+				title,
+				description,
 				product,
-				title: title,
-				description: description,
 				guides: [],
-			}
+			})
 		} else if (isHvdMetadata) {
 			const { title, description } = loadMetadata(
 				path.join(HVD_CONTENT_DIR, item)
 			)
-			const guideSlug = `/validated-designs/${pathParts.slice(0, -1).join('-')}`
-
-			// This assumes the property is already set, @TODO harden this
-			categoryGroups[categorySlug].guides = [
-				{
-					slug: guideSlug,
-					title,
-					description,
-					href: guideSlug,
-				},
-				...categoryGroups[categorySlug].guides,
-			]
+
+			// find the existing HvdCategoryGroup, because we traverse the files in order the HvdCategoryGroup YAML will always come before the HvdGuide YAML
+			const categoryGroup = hvdCategoryGroups.find(
+				(categoryGroup: HvdCategoryGroup) => {
+					return categoryGroup.slug === slug
+				}
+			)
+
+			const guideSlug = pathParts[2]
+			const categorySlug = `${slug}-${guideSlug}`
+
+			// build the category group
+			const pagesPath = path.join(product, category, guideSlug)
+			const pagesFiles = hvdRepoContents.filter((path: string) => {
+				return path.includes(pagesPath) && path.endsWith('.mdx')
+			})
+			const pages = pagesFiles.map((pagePath: string) => {
+				const pagePathParts = pagePath.split('/')
+				const pageFile = pagePathParts[pagePathParts.length - 1]
+
+				const slug = pageFile
+					.replace('.mdx', '')
+					.substring(pageFile.indexOf('-') + 1)
+
+				return {
+					slug,
+					title: slug.replaceAll('-', ' '),
+					filePath: path.join(HVD_CONTENT_DIR, pagePath),
+					href: `${basePath}/${categorySlug}/${slug}`,
+				}
+			})
+
+			categoryGroup.guides.push({
+				slug: categorySlug,
+				title,
+				description,
+				href: `${basePath}/${categorySlug}`,
+				pages,
+			})
 		}
 	})
 
+	return hvdCategoryGroups
+}
+
+export function getHvdCategoryGroupsPaths(): string[][] {
+	const categoryGroups = getHvdCategoryGroups()
+	// [[guide-slug], [guide-slug, page-slug]]
+	// e.g. [[terraform-operation-guide-adoption], [terraform-operation-guide-adoption, page-slug]]
+	const paths = []
+	for (const categoryGroup of categoryGroups) {
+		for (const guide of categoryGroup.guides) {
+			paths.push([guide.slug])
+			for (const page of guide.pages) {
+				paths.push([guide.slug, page.slug])
+			}
```

Comment:
What do you think about refactoring this to use flatMap?

Not sure if this would be the exact syntax, but something like this:
```
return categoryGroups.flatMap(categoryGroup =>
  categoryGroup.guides.flatMap(guide => [
    [guide.slug],
     ...guide.pages?.map(page => [guide.slug, page.slug])
  ])
 ```

Reply:
Ohh good point and much cleaner. Thanks @heatlikeheatwave! The only change is that the inner flatmap would need to be a map, other [guide.slug] would be removed from it's array and the structure I need for next.js static paths is:

```js
[
  [slug],
  [slug_part_1, slug_part_2]
]
```

Reply:
Ah yes, great point! Scratch the second flatMap then. Maybe something like this?
```
  return categoryGroups.flatMap(categoryGroup =>
    categoryGroup.map(({guides}, idx) => [
      [guides[idx].slug],
      ...guides[idx].pages?.map(page => [guides[idx].slug, page.slug])
    ])
  )
}
 ```